src/librustc_typeck/astconv.rs

```diff
@@ -1814,7 +1815,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             )
```

Comment:
I think here we need to look for conflicting `existential_projections` and emit an error like E0271. I think the change to `dedup` below makes it consider two projections on the same associated type but with different values to be the same.

I would also imagine that if you [try this code out](https://github.com/rust-lang/rust/issues/59326#issuecomment-646465511) you will also get a "successful" compilation.

Reply:
> I would also imagine that if you try this code out you will also get a "successful" compilation.

That code compiles, added it as a test. I still fear that the current dedup ends up being unsound though :thinking:

Reply:
Why do you think it's unsound?

Reply:
My fear was that we now `List<ExistentialPredicate>` for `dyn A + B` previously was `[Trait(A), Trait(B)]` its now just `[Trait(A)]` and I wasn't sure how we handle wf for trait objects.

I spend a bit more time looking at this and strongly believe that this PR is not concerning here
as we emit the errors before using `stable_cmp`, so anything we may "incorrectly" dedup here
already caused an error.

Added a comment summarizing my current understanding

Reply:
Huh, I guess I better look at how `stable_cmp` is defined.

Reply:
OK, yeah, this seems like not what I expected. =)

Reply:
Okay, so I've rebased this rn, I think the current state is sound, if somewhat :ghost: *spooky* :ghost: Not
sure what's the least invasive way to clean this up though...

Reply:
Sorry, I'll try to give useful feedback as soon as I can ...